src/components/events/CheckInModal/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { Modal, Typography } from '@/components/common';
+import { PublicEvent } from '@/lib/types/apiResponses';
+import Image from 'next/image';
+import style from './style.module.scss';
+
+interface CheckInModalProps {
+  open: boolean;
+  event?: PublicEvent;
+  onClose: () => void;
+}
+
+const CHECKIN_TITLES = [
+  'Yippee!',
+  'Brilliant!',
+  'Wahoo!',
+  'Awesome!',
+  'Nice to see you!',
+  'Congrats!',
+  'Hello World!',
+  'Welcome!',
+];
+
+const CheckInModal = ({ open, event, onClose }: CheckInModalProps) => {
+  if (!event) {
+    return null;
+  }
+
+  const { pointValue, title } = event;
+
+  return (
+    <Modal open={open} onClose={onClose}>
+      <div className={style.container}>
+        <div className={style.graphic}>
+          <Image src="/assets/graphics/portal/treasure.svg" alt="treasure" fill />
```

Comment:
think there could be a more  descriptive alt text, like

```suggestion
          <Image src="/assets/graphics/portal/treasure.svg" alt="A diamond with a smiley face looks at an open treasure chest." fill />
```